Fix: `capitalize` breaks with empty strings

Code reviewer: @blackwinter 

https://metafacture.org/playground/?flux=inputFile%0A%7Copen-file%0A%7Cas-lines%0A%7Cdecode-json%0A%7Cfix%28transformationFile%29%0A%7Cencode-yaml%0A%7Cprint%0A%3B&transformation=capitalize%28%22test%22%29&data=%7B%22test%22%3A+%22%22%7D

In:
```
{"test": ""}
```


Flux:

```
inputFile
|open-file
|as-lines
|decode-json
|fix(transformationFile)
|encode-yaml
|print
;
```

Error:
```
org.metafacture.metafix.FixProcessException: Error while executing Fix expression (at file:/home/playground/metafacture-playground/tmp/metafix6198042882328814353.fix, line 1): capitalize("test")
...
Caused by: java.lang.StringIndexOutOfBoundsException: begin 0, end 1, length 0
```

The function should skip the string if it is empty.

--- a/metafix/src/main/java/org/metafacture/metafix/method/field/Capitalize.java
+++ b/metafix/src/main/java/org/metafacture/metafix/method/field/Capitalize.java
@@ -35,9 +35,7 @@ public class Capitalize implements FixFunction {
 
     @Override
     public void apply(final Metafix metafix, final Record record, final List<String> params, final Map<String, String> options) {
-        if (!params.isEmpty()) {
-            record.transform(params.get(0), s -> s.substring(0, 1).toUpperCase() + s.substring(1));
-        }
+        record.transform(params.get(0), s -> s.isEmpty() ? s : s.substring(0, 1).toUpperCase() + s.substring(1));
     }
 
 }

--- a/metafix/src/main/java/org/metafacture/metafix/method/field/Capitalize.java
+++ b/metafix/src/main/java/org/metafacture/metafix/method/field/Capitalize.java
@@ -35,7 +35,9 @@ public class Capitalize implements FixFunction {
 
     @Override
     public void apply(final Metafix metafix, final Record record, final List<String> params, final Map<String, String> options) {
-        record.transform(params.get(0), s -> s.substring(0, 1).toUpperCase() + s.substring(1));
+        if (!params.isEmpty()) {
+            record.transform(params.get(0), s -> s.substring(0, 1).toUpperCase() + s.substring(1));
+        }
     }
 
 }

--- a/metafix/src/test/java/org/metafacture/metafix/MetafixMethodTest.java
+++ b/metafix/src/test/java/org/metafacture/metafix/MetafixMethodTest.java
@@ -170,6 +170,24 @@ public class MetafixMethodTest {
     }
 
     @Test
+    public void shouldSkipEmptyStringWhenCapitalize() {
+        MetafixTestHelpers.assertFix(streamReceiver, Arrays.asList(
+                "capitalize('title')"
+            ),
+            i -> {
+                i.startRecord("1");
+                i.literal("title", "");
+                i.endRecord();
+            },
+            o -> {
+                o.get().startRecord("1");
+                o.get().literal("title", "");
+                o.get().endRecord();
+            }
+        );
+    }
+
+    @Test
     public void shouldCapitalizeStringsInArray() {
         MetafixTestHelpers.assertFix(streamReceiver, Arrays.asList(
                 "capitalize('title.*')"

